tests/external_executor/executors/util.py

```diff
@@ -33,8 +53,78 @@ def terminate(self):
 
 
 @contextmanager
-def executor_thread(executor):
+def executor_thread(executor: Executor):
     et = ExecutorThread(executor)
     et.start()
     yield executor
     et.terminate()
+
+
+class ExecutorContainer:
+    def __init__(
+        self,
+        executor: str,
+        node_public_rpc_address: str,
+        network: Network,
+        supported_endpoints: Set[Tuple[str, str]],
+    ):
+        self._client = docker.DockerClient()
+
+        image_name = "ccfmsrc.azurecr.io/ccf/ci:14-02-2023-virtual"
+        LOG.info(f"Pulling image {image_name}")
+        self._client.images.pull(image_name)
+
+        # Create a container with external executor code loaded in a volume and
+        # a command to run the executor
+        command = "/workspaces/CCF/build/env/bin/python3"
+        command += " /workspaces/CCF/tests/external_executor/run_executor.py"
+        command += f' --executor "{executor}"'
+        command += f' --node-public-rpc-address "{node_public_rpc_address}"'
+        command += f' --network-common-dir "{network.common_dir}"'
+        command += f' --supported-endpoints "{",".join([":".join(e) for e in supported_endpoints])}"'
+        LOG.info(f"Creating container with command: {command}")
+        self._container = self._client.containers.create(
+            image=image_name,
+            command=command,
+            volumes={
+                "/workspaces/CCF": {
+                    "bind": "/workspaces/CCF",
+                    "mode": "rw",
+                }
+            },
+            publish_all_ports=True,
+            auto_remove=True,
+        )
+
+        LOG.info("Connecting container to network")
+        self._network = self._client.networks.get("ccf_test_docker_network")
+        self._network.connect(self._container)
+
+    def start(self):
+        LOG.info("Starting container...")
+        self._container.start()
+        time.sleep(2)
```

Comment:
To be removed?

Reply:
Removed as we now poll for registered endpoints as of https://github.com/microsoft/CCF/pull/5018/commits/90ae68730b9d7c2d62236d3188a7098bcedcfc82